bin/make-dashboard.py

```diff
@@ -23,21 +23,48 @@
     ('swcarpentry/site', 'Software Carpentry website'),
 )
 
-assert len(sys.argv) == 3, 'Usage: make-dashboard.py token-file output-file'
-token_file = sys.argv[1]
-output_file = sys.argv[2]
+USAGE = '''Usage: make-dashboard.py -o outputfile [-t tokenfile | -u username -p password]
+If no credentials are supplied, user is prompted for username and password.
+To store credentials, create a file called git-token.txt that contains your GitHub API token
+and put it in the root directory of the 'site' project.'''
 
-with open(token_file, 'r') as reader:
-    token = reader.read().strip()
+# Command-line options.
+parser = OptionParser()
+parser.add_option('-o', '--output', dest='outputfile', help='output directory')
+parser.add_option('-t', '--tokenfile', dest='tokenfile', help='GitHub token file')
+parser.add_option('-u', '--username', dest='username', help='GitHub username')
+parser.add_option('-p', '--password', dest='password', help='GitHub password')
+options, args = parser.parse_args()
 
-g = Github(token)
+assert not args, USAGE
+
+assert options.outputfile is not None, USAGE
+
+if options.tokenfile is not None:
+    assert options.username is None and options.password is None, USAGE
+    with open(options.tokenfile, 'r') as reader:
+        token = reader.read().strip()
+        g = Github(token)
+
+elif options.username is not None and options.password is not None:
+    g = Github(options.username, options.password)
+
+elif options.username is None and options.password is None:
+    username = raw_input('username: ')
+    password = raw_input('password: ')
```

Comment:
What about use

~~~
password = getpass.getpass('password: ')
~~~

to not show on the screen the password? **This will require to add `import getpass` from Python Standard Library.**